pkg/cache/db/queries.go

```diff
@@ -0,0 +1,66 @@
+package db
+
+const (
+	createTable = `
+		CREATE TABLE IF NOT EXISTS Component (
+			id INTEGER PRIMARY KEY,
+			parent_uid TEXT,
+			uid TEXT UNIQUE,
+			'group' TEXT,
+			version TEXT,
+			kind TEXT, 
+			namespace TEXT,
+			'name' TEXT,
+			health INT,
+			FOREIGN KEY(parent_uid) REFERENCES Component(uid)
+		);
+		CREATE INDEX IF NOT EXISTS idx_parent ON Component(parent_uid);
+		CREATE INDEX IF NOT EXISTS idx_uid ON Component(uid);
+	`
+
+	setComponent = `
+		INSERT INTO Component (
+			uid,
+			parent_uid,
+			'group',
+			version,
+			kind, 
+			namespace,
+			name,
+			health
+		) VALUES (
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?,
+			?
+		) ON CONFLICT(uid) DO UPDATE SET
+			parent_uid = excluded.parent_uid,
+			'group' = excluded.'group',
+			version = excluded.version,
+			kind = excluded.kind,
+			namespace = excluded.namespace,
+			name = excluded.name,
+			health = excluded.health
+	`
+
+	componentChildren = `
+		WITH RECURSIVE descendants AS (
```

Comment:
oh nvm that's the correct way to traverse for sure, didn't realize sqlite supports with recursive queries